virtualenv.py

```diff
@@ -227,7 +227,9 @@ def log(self, level, msg, *args, **kw):
                     else:
                         rendered = msg
                     rendered = ' '*self.indent + rendered
-                if hasattr(consumer, 'write'):
+                if hasattr(consumer, 'buffer'):
+                    consumer.buffer.write(rendered.encode('utf-8')+b'\n')
+                elif hasattr(consumer, 'write'):
```

Comment:
This patch is added at the early stage of this PR - now it's not necessary and I have removed it.